### PR TITLE
Like function

### PR DESCRIPTION
**Pull request by DANGHAI11**

### CODE REVIEW

File: app/Services/LikeService.php [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+            return $result;
+        } catch (Exception $ex) {
+            return false;
+        }
+    }
+
+    public function handleLike(int $postId): bool
+    {
+        try {
+            $user = User::find(Auth::id());
```
Comment (TruongPM-Regit) [inline app/Services/LikeService.php]: Auth::user()

File: app/Services/LikeService.php [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+            if ($this->statusLike($postId)) {
+                $user->likes()->detach($postId);
```
Comment (TruongPM-Regit) [inline app/Services/LikeService.php:31]: xem lại func detach

Comment (DANGHAI11) [reply app/Services/LikeService.php:31]: Bắt buộc phải có cái likes() thi mới chạy được.

File: app/Http/Controllers/LikeController.php [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    {
+        $this->likeService = $likeService;
+    }
+
+    public function like(Post $post)
+    {
+        if ($this->likeService->handleLike($post->id)) {
+            $statusLike = $this->likeService->statusLike($post->id, Auth::user());
+            $totalLike = $this->likeService->totalLikePost($post);
+            return Response::json(
```
Comment (TruongPM-Regit) [inline app/Http/Controllers/LikeController.php]: return Response::json([
'totalLike' => $totalLike,
'statusLike' => $statusLike,
], 200)

**Pull request merged**